Code/GraphMol/MolDraw2D/MolDraw2DUtils.h

```diff
@@ -75,6 +75,32 @@ RDKIT_MOLDRAW2D_EXPORT void updateDrawerParamsFromJSON(MolDraw2D &drawer,
                                                        const char *json);
 RDKIT_MOLDRAW2D_EXPORT void updateDrawerParamsFromJSON(MolDraw2D &drawer,
                                                        const std::string &json);
+
+struct ContourParams {
+  bool setScale = true;           // assumes the grid is drawn first
+  bool dashNegative = true;       // use dashed lines for negative contours
+  bool fillGrid = false;          // shade the grid
+  double gridResolution = 0.15;   // spacing between elements of the grid
+  double contourWidth = 1.0;      // linewidth for drawing contours
+  double extraGridPadding = 0.0;  // extra padding (in molecule coordinates)
+  DrawColour contourColour = {0.5, 0.5, 0.5,
+                              0.5};  // color for drawing contours
+  std::vector<DrawColour> colourMap = {
+      {0.557, 0.004, 0.322, 0.5},
+      {1, 1, 1, 0.5},
+      {0.153, 0.392, 0.098, 0.5}};  // similarity map color scheme
+};
+RDKIT_MOLDRAW2D_EXPORT void contourAndDrawGrid(
+    MolDraw2D &drawer, const double *grid, const std::vector<double> &xcoords,
+    const std::vector<double> &ycoords, size_t nContours,
+    std::vector<double> &levels, const ContourParams &ps = ContourParams());
+
+RDKIT_MOLDRAW2D_EXPORT void contourAndDrawGaussians(
```

Comment:
It looks like one can supply nContours OR levels.  I find having them in the same function is confusing.  I might have two functions:

```
void contourAndDrawGaussians( 
   MolDraw2D &drawer, const std::vector<Point2D> &locs,
    const std::vector<double> &heights, const std::vector<double> &widths,
    size_t nContours, 
    const ContourParams &ps = ContourParams());
```

```
void contourAndDrawGaussians( 
   MolDraw2D &drawer, const std::vector<Point2D> &locs,
    const std::vector<double> &heights, const std::vector<double> &widths,
    const std::vector<double> &widths,
    const ContourParams &ps = ContourParams());
```

Reply:
The idea behind having both was to allow the code to return the contour values being used if they are automatically selected.
I will add some documentation about this as well as some sanity checking in the code.